src/lib/comms/sol-lwm2m-client.c

```diff
@@ -452,14 +452,36 @@ check_authorization(struct sol_lwm2m_client *client,
     }
 
     obj_ctx = find_object_ctx_by_id(client, ACCESS_CONTROL_OBJECT_ID);
+    //If the target Object is an Access Control Object itself,
+    // the server is authorized iff it is the owner of the object instance.
+    if (obj_id == ACCESS_CONTROL_OBJECT_ID) {
+        SOL_VECTOR_FOREACH_IDX (&obj_ctx->instances, obj_instance, i) {
+            if (obj_instance->id == instance_id) {
+                r = read_resources(client, obj_ctx, obj_instance, res, 1,
+                    ACCESS_CONTROL_OBJECT_OWNER_RES_ID);
+                if (r < 0) {
+                    SOL_WRN("Could not read Access Control"
+                        " Object's [Owner ID] resource\n");
+                    goto exit_clear_1;
+                }
+
+                if (res[0].data[0].content.integer == server_id)
+                    r = 1;
+                else
+                    r = 0;
+                goto exit_clear_1;
+            }
+        }
+    }
```

Comment:
Fixed this (and added `return -EINVAL` at the end of the "normal" case as well).

Should I make a new PR or update this one?